README.md

```diff
@@ -1019,6 +1019,255 @@ Voilà! Since we updated `index.html`, you'll need to restart the dev server. Go
+              <li key={entry.id}>
+                <h2> {React.string(entry.title)} </h2>
+                {switch (entry.content) {
+                 | None => React.null
+                 | Some(content) =>
+                   <p dangerouslySetInnerHTML={"__html": content} />
+                 }}
+              </li>
+            )
+         |> React.array
+       }}
+    </ul>
+  </div>;
+```
+
+#### 11.4: Modify `fetchFeed` to support pagination
+
+Update `fetchFeed` to accept a `page` parameter and include it in the API
+request. We’ll use [labeled
+arguments](https://reasonml.github.io/docs/en/function#named-arguments) to avoid
+confusion when calling this function:
+
+```reason
+let fetchFeed = (~username, ~page) => {
+  setState(state =>
+    {
+      ...state,
+      step:
+        switch (state.step) {
+        | Loaded(r)
+        | Loading(Some(r)) => Loading(Some(r))
+        | Loading(None)
+        | Idle
+        | InvalidUsername => Loading(None)
+        },
+    }
+  );
+  module P = Js.Promise;
+  Fetch.fetch(
+    "https://gh-feed.vercel.app/api?user="
+    ++ Username.toString(username)
+    ++ "&page="
+    ++ string_of_int(page),
+  )
+  |> P.then_(response => {
+        let status = Fetch.Response.status(response);
+        if (status === 200) {
+          /* If status is OK, proceed to parse the response */
+          response
+          |> Fetch.Response.text
+          |> P.then_(text =>
+              {
+                let data =
+                  try(Ok(text |> Json.parseOrRaise |> Feed.feed_of_json)) {
+                  | Json.Decode.DecodeError(msg) =>
+                    Js.Console.error(msg);
+                    Error("Failed to decode: " ++ msg);
+                  };
+
+                switch (data) {
+                | Error(_) =>
+                  setState(state => {...state, step: Loaded(data)})
+                | Ok(data) =>
+                  setState(state => {
+                    let updatedFeed =
+                      switch (state.step) {
+                      | Loaded(Ok(feed))
+                      | Loading(Some(Ok(feed))) =>
+                        mergeFeeds(feed, data)
+                      | _ => data
+                      };
+                    {
+                      ...state,
+                      step: Loaded(Ok(updatedFeed)),
+                      currentPage: page + 1,
+                    };
+                  })
+                };
+              }
+              |> P.resolve
+            );
+        } else {
+          /* Handle non-200 status */
+          ...
+        };
+      })
+  |> ignore;
+};
+```
+
+#### 11.5: Modify render code to include scroll sentinel and handling new state
+
+We’ll add a `sentinelRef` to track the last element of the list, which will
+trigger loading more data when it comes into view.
+
+In the render code, create a new React ref to track the last element of the
+list:
+
+```reason
+    let sentinelRef = React.useRef(Js.Nullable.null);
```

Comment:
Might be worth mentioning the difference between option and Js.Nullable.null

Reply:
https://github.com/ahrefs/react-alicante-workshop/pull/9/commits/9ce1f53a5d09a982758659596887450b4fdcd737